application/aam-backend-service/src/test/kotlin/com/aamdigital/aambackendservice/export/usecase/DefaultRenderTemplateUseCaseTest.kt

```diff
@@ -0,0 +1,380 @@
+            }
+            .verifyComplete()
+    }
+
+    @Test
+    fun `should return Failure with parsed error message`() {
+        // given
+        val templateRef = DomainReference("some-id")
+        val bodyData: JsonNode = TextNode("body-data")
+        val exportTemplate = ExportTemplate(
+            id = "export-id",
+            templateId = "export-template-id",
+            title = "export-title",
+            description = "export-description",
+            applicableForEntityTypes = emptyList()
+        )
+
+        whenever(templateStorage.fetchTemplate(templateRef)).thenReturn(Mono.just(exportTemplate))
+
+        mockWebServer.enqueue(
+            MockResponse().setBody(
+                """
+            {
+                "success": false,
+                "error": "this is an error message"
+            }
+                """.trimIndent()
+            )
+        )
+
+        StepVerifier.create(
+            service.execute(
+                CreateRenderTemplateRequest(
+                    templateRef, bodyData
+                )
+            )
+        )
+            .consumeNextWith { response ->
+                // then
+                assertThat(response).isInstanceOf(UseCaseOutcome.Failure::class.java)
+
+                assertEquals("this is an error message", (response as UseCaseOutcome.Failure).errorMessage)
+            }
+            .verifyComplete()
+    }
+
+    @Test
+    fun `should return Failure when createRenderRequest returns no response`() {
+        // given
+        val templateRef = DomainReference("some-id")
+        val bodyData: JsonNode = TextNode("body-data")
+        val exportTemplate = ExportTemplate(
+            id = "export-id",
+            templateId = "export-template-id",
+            title = "export-title",
+            description = "export-description",
+            applicableForEntityTypes = emptyList()
+        )
+
+        whenever(templateStorage.fetchTemplate(templateRef)).thenReturn(Mono.just(exportTemplate))
+
+        StepVerifier.create(
+            service.execute(
+                CreateRenderTemplateRequest(
+                    templateRef, bodyData
+                )
+            )
+        )
+            .consumeNextWith { response ->
+                // then
+                assertThat(response).isInstanceOf(UseCaseOutcome.Failure::class.java)
+
+                assertThat((response as UseCaseOutcome.Failure).cause)
+                    .isInstanceOf(WebClientRequestException::class.java)
+
+                assertThat(response.cause?.cause)
+                    .isInstanceOf(NetworkException::class.java)
+
+                assertThat(response.errorCode)
+                    .isEqualTo(CreateRenderTemplateErrorCode.CREATE_RENDER_REQUEST_FAILED_ERROR)
+            }
+            .verifyComplete()
+    }
+
+    @Test
+    fun `should return Failure when fetchRenderIdRequest fails`() {
```

Comment:
where in the //given is the error mocked here? at first glance to me this all looks like it should work

Reply:
The third request should fail in this test. We just mock the first two requests, so the third one will run into an timeout.